calyx-py/test/correctness/queues/binheap/binheap.py

```diff
@@ -31,47 +31,46 @@ def insert_swap(prog, name, width, len, idx_w):
     return comp
 
 
-def insert_binheap(prog, name, queue_len_factor):
+def insert_binheap(prog, name, queue_size_factor):
     """Inserts the component `binheap` into the program.
 
     It is a minimum binary heap, represented as an array.
 
     It has:
     - three inputs, `cmd`, `value`, and `rank`.
-    - one memory, `mem`, of size `(2**queue_len_factor) - 1`.
+    - one memory, `mem`, of size `2**queue_size_factor`.
     - two ref registers, `ans` and `err`.
     """
 
     comp = prog.component(name)
 
-    max_queue_len = (2**queue_len_factor) - 1
+    max_queue_size = 2**queue_size_factor
```

Comment:
Ah so you may actually want to stick to `2**factor - 1`, since a binary tree with `2**n - 1` nodes always has a nice complete Christmas tree-like shape. The `2**n`th node, a leaf, will form a weird final fringe, no?

Reply:
We talked about this in person, but I'll record it here for the benefit of others.

The reason we do `2**n` instead of `2**n - 1` is so that it maps nicely onto the PIFOs and FIFOs. That is, we'd like setting the same `queue_size_factor` on the `binheap` and PIFO/FIFO to yield queues of the same size. Of course, we could change the PIFO/FIFO such that their sizes are `2**n - 1`, but they use tricks with overflow and pointers that rely on their size being `2**n`.